lib/ansible/modules/system/parted.py

```diff
@@ -0,0 +1,654 @@
+            'unit': {
+                'default': 'KiB',
+                'choices': [
+                    's', 'B', 'KB', 'KiB', 'MB', 'MiB', 'GB', 'GiB', 'TB',
+                    'TiB', '%', 'cyl', 'chs', 'compact'
+                ],
+                'required': False,
+                'type': 'str'
+            },
+
+            # mklabel <label-type> command
+            'label': {
+                'choices': [
+                    'aix', 'amiga', 'bsd', 'dvh', 'gpt', 'loop', 'mac', 'msdos',
+                    'pc98', 'sun', ''
+                ],
+                'required': False,
+                'type': 'str'
+            },
+
+            # mkpart <part-type> [<fs-type>] <start> <end> command
+            'part_type': {
+                'default': 'primary',
+                'choices': ['primary', 'extended', 'logical'],
+                'required': False,
+                'type': 'str'
+            },
+            'part_start': {'default': '0%', 'type': 'str', 'required': False},
+            'part_end': {'default': '100%', 'type': 'str', 'required': False},
+
+            # name <partition> <name> command
+            'name': {'type': 'str', 'required': False},
+
+            # set <partition> <flag> <state> command
+            'flags': {'type': 'list', 'required': False},
+
+            # rm/mkpart command
+            'state': {
+                'choices': ['present', 'absent'],
+                'required': False,
+                'type': 'str'
+            }
+        },
+        required_together=[
+            ['number', 'state']
+        ],
+        supports_check_mode=True,
+    )
+
+    # Data extraction
+    device      = module.params['device']
+    align       = module.params['align'].lower()
+    number      = module.params['number']
+    unit        = module.params['unit']
+    label       = module.params['label']
+    part_type   = module.params['part_type'].lower()
+    part_start  = module.params['part_start']
+    part_end    = module.params['part_end']
+    name        = module.params['name']
+    state       = module.params['state']
+    flags       = module.params['flags']
+
+    # Conditioning
+    if state:
+        state = state.lower()
+    if number and number < 0:
+        module.fail_json(msg="The partition number must be non negative.")
+
+    # Read the current disk information
+    current_device = get_device_info(device, unit)
+    current_parts = current_device['partitions']
+
+    if state and state == 'present':
+        # Default value for the label
+        if not current_device['generic']['table'] or \
+           current_device['generic']['table'] == 'unknown' and \
+           not label:
+            label = 'msdos'
+
+        # Assign label if required
+        if label:
+            script += "mklabel {0} ".format(label)
+
+        # Create partition if required
+        if part_type and not part_exists(current_parts, 'num', number):
+            script += "mkpart {0} {1} {2} ".format(
```

Comment:
Using format() here means it won't work with Python 2.6 and older. Since Ansible supports Python 2.4 and higher you should really be using the old formatting method.
See http://pyformat.info/